### PR TITLE
Remove Rescale Boolean Variable from Dropout Layer

### PR DESCRIPTION
**Pull request by dakshitagrawal97**
This PR removes the unnecessary rescale boolean variable used in the dropout layer.  The user should not be given the option of rescaling the inputs during the testing phase.  The inputs are scaled during the training phase so that the expected sum remains unchanged.

@zoq please have a review of the changes.

### CODE REVIEW

**Comment by zoq** [comment]
This looks good so far, can you take a look at the build issues?

**Comment by dakshitagrawal97** [comment]
@zoq Please have a look.  I have fixed the build errors.

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 1]
```diff
@@ -47,108 +47,104 @@ namespace ann /** Artificial Neural Network. */ {
...
->
-class Dropout
-{
+template<
+                typename InputDataType = arma::mat,
+                typename OutputDataType = arma::mat
+        >
+class Dropout {
  public:
-  /**
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/dropout.hpp]: Looks like the indentation is wrong, take a look at https://github.com/mlpack/mlpack/wiki/DesignGuidelines for more information.

Comment (dakshitagrawal97) [reply src/mlpack/methods/ann/layer/dropout.hpp]: Thanks for this reference!  So all the method declarations in public should be two spaces?

Comment (zoq) [reply src/mlpack/methods/ann/layer/dropout.hpp]: Indentation are two spaces, realistically you can take a look at the existing classes for examples.

**Comment by dakshitagrawal97** [comment]
@zoq I have fixed the indentation errors.  It should be ready to merge.  Let me know if there are any changes to be made.  :)

**Comment by dakshitagrawal97** [comment]
@zoq I am not sure why Travis Build error has occurred.  The only changes were indentations. :(

**Comment by zoq** [comment]
Don't worry it has nothing to do with the PR ``The job exceeded the maximum time limit for jobs, and has been terminated."``

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 2]
```diff
@@ -47,21 +47,18 @@ namespace ann /** Artificial Neural Network. */ {
...
  *         arma::sp_mat or arma::cube).
  */
-template <
-    typename InputDataType = arma::mat,
-    typename OutputDataType = arma::mat
->
-class Dropout
-{
+template<
+                typename InputDataType = arma::mat,
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/dropout.hpp]: Line 51 and 52, should be tabbed twice from where the previous line began, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more information.

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 3]
```diff
@@ -70,7 +67,7 @@ class Dropout
    * @param output Resulting output activation.
    */
   template<typename eT>
-  void Forward(const arma::Mat<eT>&& input, arma::Mat<eT>&& output);
+  void Forward(const arma::Mat<eT> &&input, arma::Mat<eT> &&output);
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/dropout.hpp]: Style issue; references and pointers are attached to the type.

File: src/mlpack/methods/ann/layer/dropout.hpp [hunk 4]
```diff
@@ -80,50 +77,48 @@ class Dropout
...
-  bool& Deterministic() { return deterministic; }
+  bool &Deterministic() { return deterministic; }
 
   //! The probability of setting a value to zero.
   double Ratio() const { return ratio; }
 
   //! Modify the probability of setting a value to zero.
-  void Ratio(const double r)
-  {
+  void Ratio(const double r) {
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/dropout.hpp]: Avoid placing opening braces on the same line; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement for more information.

Comment (dakshitagrawal97) [reply src/mlpack/methods/ann/layer/dropout.hpp]: I am sorry for not being thorough with the style guidelines.  It is a bit different from what I am used to, but I will make sure it does not happen.

Comment (zoq) [reply src/mlpack/methods/ann/layer/dropout.hpp]: No worries.

**Comment by sourabhvarshney111** [comment]
@dakshitagrawal97 I think just some style issues are remaining. You can do that or I would like to take this PR from here.

**Comment by zoq** [comment]
@dakshitagrawal97 @sourabhvarshney111 merged with #1354, thanks again for the contribution.

**Comment by sourabhvarshney111** [comment]
@zoq We are obliged : )

**Pull request merged**